Engine/CMakeLists.txt

```diff
@@ -163,6 +171,11 @@ endif()
 set(LIBRARY_NAME ElvenEngine)
 add_library(${LIBRARY_NAME} STATIC ${ENGINE_SOURCES})
 
+target_compile_features(${LIBRARY_NAME} PRIVATE cxx_std_20)
```

Comment:
For some reason, my clang did not catch using C++ 20 right away for this project, so I forcefully included it here